### PR TITLE
subset moidfied original dataset

### LINKED ISSUE #201: subset changes rowsize in the original dataset
In the following example the subset operation changes the values of rowsize. Why is this happening? Is this an issue with deep/shallow copy or something else?

```
ds = clouddrift.datasets.gdp1h()
display(np.sum(ds.rowsize))
#extract Gulf of mexico (gom) from 2000 to 2023
ds_gom = subset(ds, {"lat": (18, 31), "lon": (-98, -78), "time": (np.datetime64("2000-01-01"), np.datetime64("2023-01-01"))})
display(np.sum(ds.rowsize))
```
Screenshot 2023-06-25 at 4 01 52 PM

### PR DESCRIPTION
**Pull request by philippemiron**
Fix #201

### CODE REVIEW

**Review by selipot — COMMENTED** [review]
I'll leave it to @milancurcic

File: clouddrift/analysis.py [hunk 1]
```diff
@@ -987,11 +987,12 @@ def subset(ds: xr.Dataset, criteria: dict) -> xr.Dataset:
         warnings.warn("No data matches the criteria; returning an empty dataset.")
         return xr.Dataset()
     else:
-        # update rowsize
-        id_count = np.bincount(ds.ids[mask_obs])
-        ds["rowsize"].values[mask_traj] = [id_count[i] for i in ds.ID[mask_traj]]
         # apply the filtering for both dimensions
-        return ds.isel({"traj": mask_traj, "obs": mask_obs})
+        ds_sub = ds.isel({"traj": mask_traj, "obs": mask_obs})
+        # update the rowsize
```
[the remark this answers is not in the document]
Comment (philippemiron) [reply clouddrift/analysis.py:992]: This corrects the bug you described. I also added this test (which was failing before this PR)

```
def test_ds_unmodified(self):
        ds_original = self.ds.copy(deep=True)
        ds_sub = subset(self.ds, {"test": True})
        xr.testing.assert_equal(ds_original, self.ds)

```
that make sure that the original `ds` is not modified after submitting. 

The problem is that I was previously doing the following:
```
# update rowsize
id_count = np.bincount(ds.ids[mask_obs])
ds["rowsize"].values[mask_traj] = [id_count[i] for i in ds.ID[mask_traj]]

# apply the filtering for both dimensions
return ds.isel({"traj": mask_traj, "obs": mask_obs})
```
which modifies the `rowsize` on the original dataset (`ds`) before filtering with the `.isel` operation. Now I'm subsetting, creating `ds_sub`—a copy of the subset of the original dataset—and then modifying the `rowsize` of `ds_sub`.

**Comment by philippemiron** [comment]
Addressed #149 at the same time.

**Review by milancurcic — APPROVED** [review]
Great find and fix, all looks good, thanks.

**Pull request merged**